en/lessons/libraries/stream-data.md

```diff
@@ -0,0 +1,417 @@
+---
+version: 1.0.0
+title: StreamData
+---
+
+An example-based unit testing library like [ExUnit](https://hexdocs.pm/ex_unit/ExUnit.html) is a wonderful tool to help you verify that your code works the way you think it does. However, example-based unit tests have some drawbacks:
+
+* It can be easy to miss edge cases, since you're only testing a few inputs.
+* You can write these tests without thinking through your requirements thoroughly.
+* These tests can be very verbose when you use several examples for one function.
+
+In this lesson we're going to explore how [StreamData](https://github.com/whatyouhide/stream_data) can help us overcome some of these drawbacks.
+
+{% include toc.html %}
+
+## What is StreamData?
+
+[StreamData](https://github.com/whatyouhide/stream_data) is a library that performs stateless property-based testing.
+
+The StreamData library will run each test [100 times by default](https://hexdocs.pm/stream_data/ExUnitProperties.html#check/1-options), using random data each time. When a test fails, StreamData will try to [shrink](https://hexdocs.pm/stream_data/StreamData.html#module-shrinking) the input to the smallest value that causes the test failure. This can be helpful when you have to debug your code! If a 50-element list causes your function to break, and only one of the list elements is problematic, StreamData can help you identify the offending element.
+
+This testing library has two main modules. [`StreamData`](https://hexdocs.pm/stream_data/StreamData.html) generates streams of random data. [`ExUnitProperties`](https://hexdocs.pm/stream_data/ExUnitProperties.htm) lets you run tests against your functions, using the generated data as your input.
+
+You might be asking how you can say anything meaningful about a function if you don't know what your exact inputs are. Read on!
+
+## Installing StreamData
+
+First, create a new Mix project. Refer to [New Projects](https://elixirschool.com/en/lessons/basics/mix/#new-projects) if you need some help.
+
+Second, add StreamData as a dependency to your `mix.exs` file:
+
+```elixir
+defp deps do
+  [{:stream_data, "~> x.y", only: :test}]
+end
+```
+
+Just replace `x` and `y` with the version of StreamData shown in the library's [installation instructions](https://github.com/whatyouhide/stream_data#installation).
```

Comment:
This feels a little awkward to me, but I didn't want the lesson to go out of date if they released a new version of StreamData. What are your thoughts?

Reply:
Great move IMO. I've thought about this for lessons before and didn't come up with this but it works for future proofing the lesson _and_ with how many translations we have, the work on updating just that version number compounds _quite_ quickly - so this seems minor but actually will save 10s of pull requests of the event of this being updated and assuming every language did a translation PR to keep it current.